MovieApp/MovieDetailsViewController.swift

```diff
@@ -0,0 +1,351 @@
+        stackViewTop.addArrangedSubview(roleLabel0)
+        stackViewTop.addArrangedSubview(roleLabel1)
+        stackViewTop.addArrangedSubview(roleLabel2)
+        
+        stackViewBottom = UIStackView()
+        stackViewBottom.axis = .horizontal
+        stackViewBottom.alignment = .fill
+        stackViewBottom.distribution = .fillEqually
+        stackViewBottom.translatesAutoresizingMaskIntoConstraints = false
+        roleLabel3 = CustomRoleView(name: "Don Heck", role: "Screenplay")
+        roleLabel4 = CustomRoleView(name: "Jack Marcum", role: "Screenplay")
+        roleLabel5 = CustomRoleView(name: "Matt Holloway", role: "Screenplay")
+        stackViewBottom.addArrangedSubview(roleLabel3)
+        stackViewBottom.addArrangedSubview(roleLabel4)
+        stackViewBottom.addArrangedSubview(roleLabel5)
+    }
+    
+    private func addSubviews() {
+        view.addSubview(scrollView)
+        
+        scrollView.addSubview(contentView)
+        contentView.addSubview(backgroundImageView)
+        
+        
+        backgroundImageView.addSubview(scoreLabel)
+        backgroundImageView.addSubview(userScoreLabel)
+        
+        backgroundImageView.addSubview(titleLabel)
+        backgroundImageView.addSubview(yearLabel)
+        
+        backgroundImageView.addSubview(dateLabel)
+        backgroundImageView.addSubview(countryLabel)
+        
+        backgroundImageView.addSubview(genreLabel)
+        backgroundImageView.addSubview(durationLabel)
+        
+        backgroundImageView.addSubview(starGraphicView)
+        
+        
+        contentView.addSubview(overviewLabel)
+        
+        contentView.addSubview(descriptionLabel)
+        
+        contentView.addSubview(stackViewTop)
+        contentView.addSubview(stackViewBottom)
+    }
+    
+    private func setConstraints() {
+        scrollView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        contentView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+            $0.width.equalTo(view)
+        }
+        
+        backgroundImageView.snp.makeConstraints {
+            $0.bottom.equalTo(contentView.snp.centerY)
+            $0.trailing.leading.equalToSuperview()
+            $0.top.equalToSuperview()
+        }
+
+        
+        scoreLabel.snp.makeConstraints {
+            $0.width.equalTo(30)
+            $0.height.equalTo(10)
+            $0.bottom.equalTo(titleLabel.snp.top).inset(-2)
+            $0.leadingMargin.equalTo(contentView.safeAreaLayoutGuide).inset(30)
+        }
+        userScoreLabel.snp.makeConstraints {
+            $0.width.equalTo(100)
+            $0.height.equalTo(10)
+            $0.bottom.equalTo(titleLabel.snp.top).inset(-2)
+            $0.leadingMargin.equalTo(scoreLabel.snp.trailing).inset(-8)
+        }
+        
+        
+        titleLabel.snp.makeConstraints {
+            $0.width.equalTo(110)
+            $0.height.equalTo(50)
```

Comment:
In most cases, we don't need to set width or height constraints for one-line labels. For example, if the movie title was longer it would not fit the given constraints. Therefore, we just let the UILabel autoresize based on the text content.